Assets/Plugins/iOS/Shopify/PaymentAuthorizationControlling/PKPaymentAuthorizationViewController+PaymentAuthorizationControlling.swift

```diff
@@ -0,0 +1,64 @@
+//
+//  PKPaymentAuthorizationViewController+PaymentAuthorizationControlling.swift
+//  UnityBuySDK
+//
+//  Created by Shopify.
+//  Copyright © 2017 Shopify Inc. All rights reserved.
+//
+//  Permission is hereby granted, free of charge, to any person obtaining a copy
+//  of this software and associated documentation files (the "Software"), to deal
+//  in the Software without restriction, including without limitation the rights
+//  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+//  copies of the Software, and to permit persons to whom the Software is
+//  furnished to do so, subject to the following conditions:
+//
+//  The above copyright notice and this permission notice shall be included in
+//  all copies or substantial portions of the Software.
+//
+//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+//
+
+import Foundation
+import PassKit
+
+extension PKPaymentAuthorizationViewController: PaymentAuthorizationControlling {
+    
+    var authorizationDelegate: NSObjectProtocol? {
+        get {
+            return delegate
+        }
+        set {
+            if let delegate = newValue as? PKPaymentAuthorizationViewControllerDelegate {
+                self.delegate = delegate
+            } else {
+                self.delegate = nil
+            }
+        }
+    }
+    
+    func present(completion: ((Bool) -> Void)?) {
+        if let rootViewController = UIApplication.shared.delegate?.window??.rootViewController {
```

Comment:
I find long optional chains hard to read. It's up to you, but I prefer to do something like this instead:
```swift
if let window = UIApplication.shared.delegate?.window,
    let rootViewController = window.rootViewController {
    ...
}
```